### ISSUE TITLE
bug(commonjs): circular dependency detected, TSundefined

### ISSUE THREAD

**Issue by CaerusKaru** [angular/angular-cli#7697]
### Bug Report or Feature Request (mark with an `x`)
```
- [ x ] bug report -> please search issues before submitting
- [ ] feature request
```

### Versions.

```
@angular/cli: 1.5.0-beta.0
node: 8.4.0
os: linux x64
@angular/animations: 5.0.0-beta.7
@angular/cdk: 2.0.0-beta.11-2f1f0fd
@angular/cli: 1.5.0-beta.0
@angular/common: 5.0.0-beta.7
@angular/compiler: 5.0.0-beta.7
@angular/core: 5.0.0-beta.7
@angular/flex-layout: 2.0.0-beta.9-5f198a3
@angular/forms: 5.0.0-beta.7
@angular/http: 5.0.0-beta.7
@angular/material: 2.0.0-beta.11-2f1f0fd
@angular/platform-browser: 5.0.0-beta.7
@angular/platform-browser-dynamic: 5.0.0-beta.7
@angular/platform-server: 5.0.0-beta.7
@angular/router: 5.0.0-beta.7
@angular/app-shell: 0.1.0
@angular/compiler-cli: 5.0.0-beta.7
@angular/language-service: 5.0.0-beta.7
@angular/service-worker: 1.0.0-beta.16
typescript: 2.4.2
```

### Repro steps.

Try to build a commonjs bundle. The build will finish, but the bundle will be several factors of size too small. The exact build command used:

```
ng build --prod --build-optimizer --app universal --output-hashing=none
```

### The log given by the failure.

```
ERROR in error TSundefined: 
```

See comment below for typescript v2.5.2 error log

### Desired functionality.

Build should complete successfully as it did before 1.5.0-beta.0

**Comment by CaerusKaru** [comment]
Upgraded to node v8.5.0 with no effect.

**Comment by LukasMarx** [comment]
Having the same issue. Just updated to angular 5.0 beta.7 and angular-cli 1.5.0-beta.0. Trying to build an universal bundle, as well. 

`ng build --prod --app 1`

Error also only occurs when building with the --aot/prod flag. Normal builds run just fine. Unfortunately, aot is required for server side usage. 

Not using build-optimizer (as used above) does not resolve the issue.

**Comment by LukasMarx** [comment]
Turned out, my typescript version was below the requirements (2.3.x). Updating to the newest version 2.5.2 seemed to solve the problem. Not quite sure though, as I get another error now.

**Comment by CaerusKaru** [comment]
To clarify, the error for typescript v2.5.2 is a circular dependency error, where one does not exist.

Example is below. The context is that landing is a lazy-loaded module, landing-help is a component in the module. There is no such warning generated for the standard non-commonjs build.
```
src/app/landing/landing-help/landing-help.component.ngfactory.js -> src/app/landing/landing-help/landing-help.component.ts -> src/app/landing/landing.module.ngfactory.js -> src/app/landing/landing-help/landing-help.component.ngfactory.js
```

**Comment by CaerusKaru** [comment]
This issue has been resolved for me with the release of 1.5.0-beta.2. I also updated angular to 5.0.0-rc.0. If anyone is still encountering this issue after following these upgrade steps, I will reopen the issue.
